pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -167,55 +169,89 @@ func handleError(message string, listOfErrors ...error) error {
 	return fmt.Errorf("errors: %s, msg: %s", strings.Join(errSlice, ", "), message)
 }
 
+// ValidateBuildRegistration verifies that a referenced Build is properly register
+func (r *ReconcileBuildRun) ValidateBuildRegistration(ctx context.Context, build *buildv1alpha1.Build, buildRun *buildv1alpha1.BuildRun) error {
+	if build.Status.Registered != corev1.ConditionTrue {
+		err := fmt.Errorf("The Build is not registered correctly, registered status: %s, reason: %s", build.Status.Registered, build.Status.Reason)
+		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+		return handleError("Build is not ready", err, updateErr)
+	}
+	return nil
+}
+
+// GetBuildRunObject retrieves an existing BuildRun based on a name and namespace
+func (r *ReconcileBuildRun) GetBuildRunObject(ctx context.Context, objectName string, objectNS string, buildRun *buildv1alpha1.BuildRun) error {
+	if err := r.client.Get(ctx, types.NamespacedName{Name: objectName, Namespace: objectNS}, buildRun); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBuildObject retrieves an existing Build based on a name and namespace
+func (r *ReconcileBuildRun) GetBuildObject(ctx context.Context, objectName string, objectNS string, build *buildv1alpha1.Build) error {
+	if err := r.client.Get(ctx, types.NamespacedName{Name: objectName, Namespace: objectNS}, build); err != nil {
+		return err
+	}
+	return nil
+}
+
+// VerifyRequestName parse a Reconcile request name and looks for an associated BuildRun name
+// If the BuildRun object exists, it will update it with an error.
+func (r *ReconcileBuildRun) VerifyRequestName(ctx context.Context, request reconcile.Request, buildRun *buildv1alpha1.BuildRun) {
+	// Check if the name belongs to a TaskRun generated name https://regex101.com/r/Wjs3bV/9
+	matched, _ := regexp.MatchString(generatedNameRegex, request.Name)
```

Comment:
Here you then can call compiledRegexp.MatchString

Reply:
done